arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,389 @@
+            }
+        }
+    }
+    
+    // MARK: UITableViewDelegate
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath)
+        tableView.deselectRow(at: indexPath, animated: true)
+        if cell == attributesCell {
+            // Create the attribute labels.
+            if attributeLabels.isEmpty {
+                attributes?.forEach { (attribute) in
+                    attributeLabels.append(attribute.name)
+                }
+            }
+            // Prompt attribute selection.
+            let optionsViewController = OptionsTableViewController(labels: attributeLabels, selectedIndex: attributes!.count) { (index) in
+                self.selectedAttribute = self.attributes![index]
+                self.attributeLabel?.text = self.selectedAttribute?.name
+            }
+            optionsViewController.title = "Attributes"
+            show(optionsViewController, sender: self)
+        } else if cell == comparisonCell {
+            // Prompt comparison operator selection.
+            let optionsViewController = OptionsTableViewController(labels: comparisonsStrings, selectedIndex: comparisonsStrings.count) { (index) in
+                self.selectedComparison = self.comparisons[index]
+                self.comparisonLabel?.text = self.comparisonsStrings[index]
+            }
+            optionsViewController.title = "Comparison"
+            show(optionsViewController, sender: self)
+        } else if cell == valueCell {
+            if selectedAttribute != nil {
+                if let domain = selectedAttribute?.domain as? AGSCodedValueDomain {
+                    // Get the value labels.
+                    if valueLabels.isEmpty {
+                        domain.codedValues.forEach { (codedValue) in
+                            valueLabels.append(codedValue.name)
+                        }
+                    }
+                    // Prompt value selection.
+                    let optionsViewController = OptionsTableViewController(labels: valueLabels, selectedIndex: domain.codedValues.count) { (index) in
+                        self.valueLabel?.text = self.valueLabels[index]
+                        self.selectedValue = domain.codedValues[index]
+                    }
+                    optionsViewController.title = "Value"
+                    show(optionsViewController, sender: self)
+                } else {
+                    // Prompt the user to create a custom value if none are available to select.
+                    addCustomValue()
+                }
+            } else {
+                // If no attribute is selected, present an alert.
+                presentAlert(title: "No attribute selected", message: "Please select an attribute before selecting a value.")
+            }
+        } else if cell == addConditionButton {
+            addCondition()
```

Comment:
The design of the samples UI is rather interesting. Once a condition is added, there's no way to update and/or remove it?

One idea is to turn the Define New Condition section into just a Conditions section. Tapping "Add Condition" would present a new view for configuring the condition to create. Upon creation of the new condition, a new row is added to the Conditions section. That row could then be deleted (via swipe-to-delete) to remove the condition. That is the more common way of doing this sort of thing and it would simplify the sample's main view controller significantly. Something to think about.

Reply:
I agree. Maybe consider refactoring this sample a bit. I'll discuss with Viv to see the best approach.